Woof/WoofTests/ViewModel/LoginViewModelTests/LoginViewModelSetUserRoleTests.swift

```diff
@@ -0,0 +1,39 @@
+import XCTest
+
+final class LoginViewModelSetUserRoleTests: XCTestCase {
+    private var userPreferencesStorage = KeyValueStorage.testPreferencesStorage
+    private let testLoginViewModel = LoginViewModel()
+
+    override func setUp() {
+        super.setUp()
+        userPreferencesStorage.deleteData(for: KeyValueStorage.Key.userPreferences)
+    }
+
+    func testMethodExistsInAPI() {
+        testLoginViewModel.set(.sitter)
+    }
+
+    func testSetsEveryPossibleRole() {
+        Role.allCases.forEach {
+            // Given When
+            testLoginViewModel.set($0)
```

Comment:
nit:

I'd recommend to use the the "full" constant name instead of the `$0` syntax to improve readability (in case where the closure is multiline):

```swift
func testSetsEveryPossibleRole() {
    Role.allCases.forEach { role in
        // Given
        let expectedSavedRole = role
        
        // When
        testLoginViewModel.set(role)
             
        // Then
        XCTAssertEqual(PreferencesHandler.getUserRole(), expectedSavedRole)
    }
}
```

Reply:
thx
fixed